engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnCompensationSubscriptionBehaviour.java

```diff
@@ -133,6 +139,33 @@ public void deleteNotTriggeredSubscriptions(final BpmnElementContext context) {
                 compensation.getRecord()));
   }
 
+  public void createCompensationSubscriptionForSubprocess(
+      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
+    if (hasCompletedActivityWithCompensationHandler(element, context)) {
+      final var key = keyGenerator.nextKey();
+      final var compensation =
+          new CompensationSubscriptionRecord()
+              .setTenantId(context.getTenantId())
+              .setProcessInstanceKey(context.getProcessInstanceKey())
+              .setProcessDefinitionKey(context.getProcessDefinitionKey())
+              .setCompensableActivityScopeId(BufferUtil.bufferAsString(element.getId()))
+              .setSubprocessSubscription(true);
```

Comment:
For finding the subprocess maybe not, but in this way is easier to separate the subscription related to compensation handler and subscription related to subprocesses. Otherwise all the time I need to filter out not valid record, see an example [here](https://github.com/camunda/zeebe/blob/6dd788a1323d928045ba3fee074febbebc8efff2/engine/src/main/java/io/camunda/zeebe/engine/state/compensation/DbCompensationSubscriptionState.java#L123-L125)

Reply:
I think it is not necessary to filter the subscriptions. Only when triggering the compensation.  

We could replace `findSubscriptionsByCompensableActivityScopeId()` by getting all subscriptions and filtering directly using Java's stream API.